internals/maintenance_scripts.py

```diff
@@ -539,10 +534,14 @@ def handle_activation(self, stage: Stage, stage_dict: StageDict) -> None:
       origin_trials_client.activate_origin_trial(stage.origin_trial_id)
       cloud_tasks_helpers.enqueue_task(
           '/tasks/email-ot-activated', {'stage': stage_dict})
+      stage.ot_setup_status = OT_ACTIVATED
     except requests.RequestException:
       cloud_tasks_helpers.enqueue_task(
           '/tasks/email-ot-activation-failed', {'stage': stage_dict})
-
+      stage.ot_setup_status = OT_ACTIVATION_FAILED
+      # The activation still needs to occur,
+      # so the activation date is set for current date.
+      stage.ot_activation_date = date.today()
```

Comment:
I think at this point, it would require investigations, otherwise it will be retried repeatedly?

Reply:
Yes, it will require investigation, and the request should not be retried again based on the logic. The date will not necessarily be used for the activation script if the activation failed here, but I thought it a good idea to set the activation date for any trial that still needs activating 😅